test/fail_compilation/b19730.d

```diff
@@ -0,0 +1,11 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/b19730.d(9): Error: found `)` while expecting `=` or identifier
+---
+*/
+void func() {
+  bool x;
+  if (const x) {}
+  if (const size_t a = 0) {}
```

Comment:
That should not be in `fail_compilation`

Reply:
The entire failure text must be matched anyway so I don't think thats a problem, although it could do with a comment stating that it is expected to generate no error.

Reply:
It is because it's not the first error message, and while the compiler does a good job at showing most messages, there's no guarantee it will show them all.

Reply:
If the compiler emits too many errors, it will output additional text saying error limit reached, which would have to be matched.

Reply:
added in a compilable test case